resources/aws_ec2.go

```diff
@@ -829,9 +836,29 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
 		return
 	}
-	post, _ := ioutil.ReadAll(req.Body)
-	if obj.debug {
-		log.Printf("%s: Post: %s", obj, string(post))
+	// decode json
+	decoder := json.NewDecoder(req.Body)
+	var post postData
+	err := decoder.Decode(&post)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
```

Comment:
If we didn't decode the message, we shut it all down? I'm okay with that, assuming aws doesn't sent erroneous messages often, but we should probably log some message to let us know their API changed...